src/startup_coordinator.c

```diff
@@ -55,6 +57,26 @@ static lf_ret_t StartupCoordinator_connect_to_neighbors_blocking(StartupCoordina
   return LF_OK;
 }
 
+void StartupCoordinator_schedule_timers_joining(StartupCoordinator *self, Reactor *reactor,
+                                                interval_t federation_start_time, interval_t join_time) {
+  lf_ret_t ret;
+  for (size_t i = 0; i < reactor->triggers_size; i++) {
+    Trigger *trigger = reactor->triggers[i];
+    if (trigger->type == TRIG_TIMER) {
+      Timer *timer = (Timer *)trigger;
+      const interval_t duration = join_time - federation_start_time - timer->offset;
```

Comment:
Should this be adding the timer->offset?

Reply:
No I dont think so here, I am calculating the time since the first firing of this time and for that I need to subtract the offset. 

I am adding the offset back two lines down when I calculate the next firing time.